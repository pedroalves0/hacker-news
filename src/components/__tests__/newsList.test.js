import React from 'react';
import { shallow } from 'enzyme';
import NewsList from '../newsList/newsList';

describe('components/newsList', () => {
    const items = [
        { 
            id: 10,
            by: 'author',
            time: 1512157581,
            title: 'title',
            url: 'http://www.url.com/url/url/'
        },
        { 
            id: 20,
            by: 'author2',
            time: 1512157582,
            title: 'title2',
            url: 'http://www.url2.com/url2/url2/'
        }
    ];

    test('should render correctly', () => {
        const wrapper = shallow(
            <NewsList
                isFetching={false}
                isOffline={false}
                items={items}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render correctly when fetching', () => {
        const wrapper = shallow(
            <NewsList
                isFetching={true}
                isOffline={false}
                items={items}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});