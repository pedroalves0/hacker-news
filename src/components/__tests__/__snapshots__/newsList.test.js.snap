// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/newsList should render correctly 1`] = `
<InfiniteScroller
  isFetching={false}
  isOffline={false}
  onScrollBottomReached={[Function]}
>
  <ul
    className="newslist"
  >
    <NewsListItem
      by="author"
      id={10}
      key="10"
      time={1512157581}
      title="title"
      url="http://www.url.com/url/url/"
    />
    <NewsListItem
      by="author2"
      id={20}
      key="20"
      time={1512157582}
      title="title2"
      url="http://www.url2.com/url2/url2/"
    />
  </ul>
</InfiniteScroller>
`;

exports[`components/newsList should render correctly when fetching 1`] = `
<InfiniteScroller
  isFetching={true}
  isOffline={false}
  onScrollBottomReached={[Function]}
>
  <ul
    className="newslist"
  >
    <NewsListItem
      by="author"
      id={10}
      key="10"
      time={1512157581}
      title="title"
      url="http://www.url.com/url/url/"
    />
    <NewsListItem
      by="author2"
      id={20}
      key="20"
      time={1512157582}
      title="title2"
      url="http://www.url2.com/url2/url2/"
    />
    <li
      className="newslist-loader"
    >
      <onlyUpdateForKeys(Loader)
        color="#000000"
        loading={true}
        margin="2px"
        size={10}
      />
    </li>
  </ul>
</InfiniteScroller>
`;
